src/main/xerus/monstercat/downloader/TabDownloader.kt

```diff
@@ -388,18 +395,54 @@ class TabDownloader: VTab() {
 		
 		logger.trace("Setting download button text to $text")
 		button.text = text
-		button.isDisable = hasGold && !valid
+		button.isDisable = (hasGold && !valid) || !login
 		button.tooltip = Tooltip(if(hasGold) "Click to start downloading the selected Tracks" else "Click to re-check you connect.sid")
-		updateDownloadButtonAction(button, valid)
+		updateDownloadButtonAction(button, valid, login)
+	}
+
+	private fun openLoginDialog(){
+		val parent = VBox()
+		val stage = App.stage.createStage("Login to Monstercat.com", parent)
+		val emailField = TextField("").apply {
+			promptText = "Email address"
+		}
+		val passwordField = PasswordField().apply {
+			promptText = "Password"
+		}
+		val controls = HBox().apply {
+			addButton("Login") {
+				val login = APIConnection.login(emailField.text, passwordField.text)
+				logger.debug("Monstercat connection with ${emailField.text} - ${passwordField.text} returned $login")
+				if (login){
+					stage.close()
+				}else{
+					parent.children.add(parent.children.lastIndex - 1, Label("Wrong username/password"))
+					stage.sizeToScene()
+				}
+			}
+			addButton("Cancel") {
+				stage.close()
+			}
+		}
+		parent.children.addAll(emailField, passwordField, controls)
+		stage.isAlwaysOnTop = true
+		stage.focusedProperty().addListener { _, onHidden, _ ->
```

Comment:
Please reformat this file properly

Reply:
f3800ca3a21844a4ca20fc8d25aea86c4095ca04